storage/protocol_state.go

```diff
@@ -0,0 +1,18 @@
+package storage
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/badger/transaction"
+)
+
+// ProtocolState represents persistent storage for protocol state entries.
+type ProtocolState interface {
+	// StoreTx allows us to store protocol state as part of a DB tx, while still going through the caching layer.
+	StoreTx(id flow.Identifier, protocolState *flow.ProtocolStateEntry) func(*transaction.Tx) error
+	// Index indexes the protocol state by block ID.
+	Index(blockID flow.Identifier, protocolStateID flow.Identifier) func(*transaction.Tx) error
+	// ByID returns the protocol state by its ID.
```

Comment:
Should document `ErrNotFound` for this and `ByBlockID`

Reply:
👉 https://github.com/onflow/flow-go/pull/4559/commits/bd549897f434e5d8057dcd345422d047d808063f